pep-0639.rst

```diff
@@ -1,543 +1,2209 @@
+distribution at that same relative path.
 
-License Expression syntax
-'''''''''''''''''''''''''
+The **root license directory** is defined to be the project root directory
+for source trees and source distributions, and the ``license_files``
+subdirectory of the directory containing the core metadata (i.e. the
+``.dist-info`` directory containing the ``METADATA`` file) for built
+distributions and installed projects.
 
-A license expression is a string using the SPDX license expression syntax as
-documented in the SPDX specification [#spdx]_ using either Version 2.2
-[#spdx22]_ or a later compatible version. SPDX is a working group at the Linux
-Foundation that defines a standard way to exchange package information.
+The specified relative path MUST be consistent between project source trees,
+source distributions (sdists), built distributions (wheels) and installed
+projects. Therefore, inside the root license directory, packaging tools
+MUST reproduce the directory structure under which the
+source license files are located relative to the project root.
 
-When used in the ``License`` field and as a specialization of the SPDX license
-expression definition, a license expression can use the following license
-identifiers:
+Path separators MUST be the forward slash character (``/``),
+and parent directory indicators (``..``) MUST NOT be used.
+License file content MUST be UTF-8 encoded text.
 
-- any SPDX-listed license short-form identifiers that are published in the SPDX
-  License List [#spdxlist]_ using either Version 3.10 or any later compatible
-  version. Note that the SPDX working group never removes any license
-  identifiers: instead they may choose to mark an identifier as "deprecated".
+Build tools MAY and publishing tools SHOULD produce an informative warning
+if a built distribution's metadata contains no ``License-File`` entries,
+and publishing tools MAY but build tools MUST NOT raise an error.
 
-- the ``LicenseRef-Public-Domain`` and ``LicenseRef-Proprietary`` strings to
-  identify licenses that are not included in the SPDX license list.
+For all newly-uploaded distribution packages that include one or more
+``License-File`` fields and declare a ``Metadata-Version`` of ``2.3`` or
+higher, PyPI SHOULD validate that the specified files are present in all
+uploaded distributions, and MUST reject uploads that do not validate.
 
-When processing the ``License`` field to determine if it contains a valid
-license expression, tools:
 
-- SHOULD report an informational warning if one or more of the following
-  applies:
+Deprecate ``License`` field
+'''''''''''''''''''''''''''
 
-  - the field does not contain a license expression,
+The legacy unstructured-text ``License`` field is deprecated and replaced by
+the new ``License-Expression`` field.
 
-  - the license expression syntax is invalid,
+Build and publishing tools MUST raise an error if both fields are present and
```

Comment:
Does “both” here mean `License` and `License-Expression`? Needs clarification.

Reply:
Good question. Actually, the sentence immediately preceding this one (which might not have been so easy to notice viewing the diff) states,

> The legacy unstructured-text ``License`` field is deprecated and replaced by the new ``License-Expression`` field.

Given this is the very first sentence in the section, and "Build and publishing tools MUST raise an error if both fields are present" is the second, I'm a little unsure how a reader might plausibly conclude that "both fields" was referring to anything other than those same two fields. However, for the avoidance of doubt, I inserted "these" between "both" and "fields" to very explicitly refer to them.